### PR TITLE
Add register page

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: app/elements/login-page-email/login-page-email.html [hunk 1]
```diff
@@ -67,6 +67,11 @@
 				<paper-input type="text" label="Username or e-mail" value="{{email}}" autofocus="true" on-keydown="onKeydown"></paper-input>
 				<paper-button class$="{{empty}}" flat colorful on-change="checkEmpty" on-click="next">Next</paper-button>
 			</div>
+			<div class="horizontal layout">
+				<span class="flex"></span>
+				<a href="/register">Don`t have an account? Sign up here.</a>
```
Comment (lorenzleutgeb) [inline app/elements/login-page-email/login-page-email.html]: Please use ' not `

File: app/elements/register/register-page.html [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	<template>
+		<user-service id="userService" user={{savedUser}}></user-service>
+		<paper-material elevation="1">
+			<div>
+				<div>
+					<iron-icon class="huge" icon="account-circle"></iron-icon>
+				</div>
+			</div>
+			<div>
+        <paper-input type="text" label="Name <e-mail>" value="{{user.Address}}" autofocus="true" on-keydown="onKeydown"></paper-input>
```
Comment (lorenzleutgeb) [inline app/elements/register/register-page.html]: Indent correctly

File: app/elements/routing.html [hunk 3]
```diff
@@ -120,7 +125,17 @@
 			}
 		} else {
 			app.route = 'login';
-			Polymer.Base.importHref('/elements/login-wizard/login-wizard.html');
+			Polymer.Base.importHref('/elements/login-wizard/login-wizard.html', function(){
+				if(!!ctx.querystring){
```
Comment (lorenzleutgeb) [inline app/elements/routing.html]: OMFG please return early.

File: app/elements/routing.html [hunk 4]
```diff
@@ -120,7 +125,18 @@
 			}
 		} else {
 			app.route = 'login';
-			Polymer.Base.importHref('/elements/login-wizard/login-wizard.html');
+			Polymer.Base.importHref('/elements/login-wizard/login-wizard.html', function(){
+				if(!!!ctx.querystring){
```
Comment (lorenzleutgeb) [inline app/elements/routing.html]: Why not `!!!!!`?

Comment (victorbalan) [inline app/elements/routing.html]: becasue that would be `!!!`

Comment (lorenzleutgeb) [inline app/elements/routing.html]: And that would be `!`.

File: app/elements/submission/tasks/canvas-game-task.html [hunk 5]
```diff
@@ -69,6 +69,11 @@
 	</style>
 	<template>
 		<iron-ajax
+			id="getCodeRequest"
+			handle-as="text"
```
Comment (lorenzleutgeb) [inline app/elements/submission/tasks/canvas-game-task.html]: `handle-as="json"` will spare you the `JSON.parse` in `onTemplateResponse`, just saying.

**Pull request merged**